packages/nx/src/command-line/import/utils/prepare-source-repo.ts

```diff
@@ -0,0 +1,137 @@
+import * as createSpinner from 'ora';
+import { basename, dirname, join, relative } from 'path';
+import { GitRepository } from '../../../utils/git-utils';
+import { promisify } from 'util';
+import { copyFile, mkdir, rm } from 'fs';
+
+const rmAsync = promisify(rm);
+const copyFileAsync = promisify(copyFile);
+const mkdirAsync = promisify(mkdir);
+
+export async function prepareSourceRepo(
+  gitClient: GitRepository,
+  ref: string,
+  source: string,
+  relativeDestination: string,
+  tempImportBranch: string,
+  sourceRemoteUrl: string
+) {
+  const spinner = createSpinner().start(
+    `Fetching ${ref} from ${sourceRemoteUrl}`
+  );
+  await gitClient.addFetchRemote('origin', ref);
+  await gitClient.fetch('origin', { depth: 1, ref });
```

Comment:
Check if all the history if `depth: 1`

Reply:
Yup, that messed it up. I made it the full history... 

If it's a local path... it kind of adopts the depth of the local path... so if people have a shallow clone, then they might be missing some commits :thinking: 

But if it's a remote path, we get the full history, however long it is